### DISCUSSION TITLE
madara hasher architecture

### DISCUSSION THREAD

**Discussion by tdelabro** [keep-starknet-strange/madara#1538] [General]
Right now the `ComputeTransactionHash` trait is generic over a `HasherT` type. Those are sizeless types, representing either Pedersen or Poseindon hashers. The idea was that each chain could choose which one to use for performance reasons.

The situation now is that:
- tx hash computation has been moved from the runtime to the client, as the extrinsics now receive Blockifier transactions types, wich contains the tx_hash already filled.
- the starknet spec are that txV3 use Poseidon while previous tx types keep using pedersen.

So the hasher is not a config of the chain, but one of the tx.
Would you all agree if we remove chain level hasher config, and implement the starknet design instead?

Other chains could still be able to diverge from starknet design by ignoring our implementations of  `ComputeTransactionHash` for blockifiers tx types and crate their own

**Comment by apoorvsadana** [comment]
Ya if we're moving hash calculation outside the runtime it makes sense to remove it from the Pallet config. So if I understand it correctly, in the new flow we receive `starknet-rs` types as input and convert it to the blockifier type within the client side itself?

**Reply by tdelabro** [reply]
yes
